packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/BaseModelSqlv2.ts

```diff
@@ -1424,6 +1425,42 @@ class BaseModelSqlv2 {
         case 'LinkToAnotherRecord':
         case 'Lookup':
           break;
+        case 'QrCode': {
+          const qrCodeColumn = await column.getColOptions<QrCodeColumn>();
+          const qrValueColumn = await Column.get({
+            colId: qrCodeColumn.fk_qr_value_column_id,
+          });
+
+          // If the referenced value cannot be found: cancel current iteration
+          if (qrValueColumn == null) {
+            break;
+          }
+
+          switch (qrValueColumn.uidt) {
+            case UITypes.Formula:
+              {
+                const formula =
+                  await qrValueColumn.getColOptions<FormulaColumn>();
+                if (formula.error) continue;
+                const selectQb = await formulaQueryBuilderv2(
+                  formula.formula,
+                  null,
+                  this.dbDriver,
+                  this.model
+                );
+                qb.select({
+                  [column.column_name]: selectQb.builder,
+                });
+              }
+              break;
+            default: {
+              qb.select({ [column.column_name]: qrValueColumn.column_name });
+              break;
```

Comment:
currently wrapping around my head more into the concepts of BaseModelSqlv2.ts (plus formula query builder).

Reply:
The `formulaQueryBuilder` generates an equivalent SQL query for the formula and `BaseModelSqlv2.ts` is used for all the data CRUD operations(not metadata)

Reply:
yes, that overall purpose i got and I implemented this PR with a feeling of understanding 70% of the code of these two files - I think it's good trying to go close to 90% now though (also for future tasks) :D

Reply:
I extracted out now the query builder logic for the formula and reused both for the formula on level-1 (when the actual column is a formula) as well as for when the referenced column of a qr code is a formula. 
I think simply calling the whole selectObject method wouldn't work because the handling is bit different (in case of qr code it's currently on purpose ignoring certain types for the referenced column and also the select statement irt aliasing is a bit different). 
Does that make se

Reply:
With selectObject we can only take care of normal columns, formula, and rollup columns. Not possible to extract LTAR and Lookup with the above scenario and a better approach is to handle it like an alias similar to Lookup. I will have a look :+1: .

Reply:
You can do something like follows - https://github.com/nocodb/nocodb/commit/68ab5cc27b6df3ad53c3efb9a03eb0cf1490d53e

Just setting an alias to the column so while resolving it will extract from referencing column we don't have to duplicate the query.

Reply:
As per our discussion via chat, we will keep the currently implemented approach for now and looking into moving the fetch logic to `getProto` etc (similar to resolving columns like Lookup and LTAR) later on